addons/stock_account/models/product.py

```diff
@@ -226,8 +226,15 @@ def _get_value_from_lots(self):
         return sum(lots.mapped('total_value'))
 
     def _with_valuation_context(self):
+        self_with_context = self
         valued_locations = self.env['stock.location'].search([('is_valued_internal', '=', True)])
-        return self.with_context(location=valued_locations.ids)
+        self_with_context = self.with_context(location=valued_locations.ids)
+        # In FIFO, the stack in on stock.move and their value is already computed base on the owner
+        if self.cost_method != 'fifo':
+            self_with_context = self_with_context.with_context(
```

Comment:
Can't we use `with_context(owner_id=False)`?
I don't know if it makes any sense to set your own company as owner. But this wouldn't work if setting a child partner either. :thinking:

Reply:
It was protected by a test when user use the company as an owner (so the good still valued). I don't know if we should just forbid it but currently I adapted the code